server/src/main/java/org/elasticsearch/index/engine/LuceneChangesSnapshot.java

```diff
@@ -95,14 +95,15 @@
             }
         };
         this.mapperService = mapperService;
-        this.searchBatchSize = searchBatchSize;
+        final long requestingSize = (toSeqNo - fromSeqNo) == Long.MAX_VALUE ? Long.MAX_VALUE : (toSeqNo - fromSeqNo + 1L);
```

Comment:
`(toSeqNo - fromSeqNo) == Long.MAX_VALUE` this would practically always evaluate to `false`?